ambry-replication/src/main/java/com.github.ambry.replication/ReplicaThread.java

```diff
@@ -183,7 +237,8 @@ void replicate(List<List<RemoteReplicaInfo>> replicasToReplicate) {
 
       List<RemoteReplicaInfo> activeReplicasPerNode = new ArrayList<RemoteReplicaInfo>();
       for (RemoteReplicaInfo remoteReplicaInfo : replicasToReplicatePerNode) {
-        if (!remoteReplicaInfo.getReplicaId().isDown()) {
+        ReplicaId replicaId = remoteReplicaInfo.getReplicaId();
+        if (!replicationDisabledPartitions.contains(replicaId.getPartitionId()) && !replicaId.isDown()) {
```

Comment:
Can we do this in a more atomic fashion. As per this line, some disabled might be considered in this iteration and some might be considered only in next iteraiton. Can we do this instead: for a single iteration, consider all the newly disabled partitions or not. If not, consider only in next iteration.